tests/components/cert_expiry/test_init.py

```diff
@@ -107,3 +114,26 @@ async def test_unload_config_entry(mock_now, hass):
     await hass.async_block_till_done()
     state = hass.states.get("sensor.cert_expiry_timestamp_example_com")
     assert state is None
+
+
+async def test_delay_load_during_startup(hass):
+    """Test delayed loading of a config entry during startup."""
+    hass.state = CoreState.not_running
+
+    entry = MockConfigEntry(domain=DOMAIN, data={CONF_HOST: HOST, CONF_PORT: PORT})
+    entry.add_to_hass(hass)
+
+    timestamp = future_timestamp(100)
+    with patch(
+        "homeassistant.components.cert_expiry.get_cert_expiry_timestamp",
+        return_value=timestamp,
+    ):
+        assert await async_setup_component(hass, DOMAIN, {}) is True
```

Comment:
After setting up the integration and blocking till done I'd assert that there's no sensor state. Then fire the event, or just start the core, and assert that there's a state set.

Reply:
Done. It seems that `hass.async_start()` is allowed during tests, or at least I haven't run into any negative side effects.